scripts/globals/battlefield.lua

```diff
@@ -68,10 +70,788 @@ xi.battlefield.dropChance =
     EXTREMELY_HIGH = 140,
 }
 
+xi.battlefield.id =
+{
+    FLAMES_FOR_THE_DEAD = 640,
+    BROTHERS = 643,
+    HOLY_COW = 644,
+    SHADOW_LORD_BATTLE = 160,
+    WHERE_TWO_PATHS_CONVERGE = 161,
+    SE_APOLLYON = 1293,
+}
+
+Battlefield = setmetatable({}, { __index = Container })
+Battlefield.__index = Battlefield
+Battlefield.__eq = function(m1, m2)
+    return m1.id == m2.id
+end
+
+-- Creates a new Battlefield interaction
+-- Data takes the following keys:
+--  - zoneId: Which zone this battlefield exists within (required)
+--  - battlefieldId: Battlefield ID used in the database (required)
+--  - maxPlayers: Maximum number of players allowed to enter (required)
+--  - timeLimit: Time in seconds alotted to complete the battlefield before being ejected. (required)
+--  - index: The index of the battlefield within the zone. This is used to communicate with the client on which menu item this battlefield is. (required)
+--  - levelCap: Level cap imposed upon the battlefield. Defaults to 0 - no level cap. (optional)
+--  - area: Some battlefields has multiple areas (Burning Circles) while others have fixed areas (Apollyon). Set to have a fixed area. (optional)
+--  - entryNpc: The name of the NPC used for entering
+--  - exitNpc: The name of the NPC used for exiting
+--  - allowSubjob: Determines if character subjobs are enabled or disabled upon entry. Defaults to true. (optional)
+--  - hasWipeGrace: Grants players a 3 minute grace period on a full wipe before ejecting them. Defaults to true. (optional)
+--  - canLoseExp: Determines if a character loses experience points upon death while inside the battlefield. Defaults to true. (optional)
+--  - delayToExit: Amount of time to wait before exiting the battlefield. Defaults to 5 seconds. (optional)
+--  - requiredItems: Items required to be traded to enter the battlefield.
+--                   Needs to be in the format of { itemid, quantity, useMessage = ID.text.*, wearMessage = ID.text.*, wornMessage = ID.text.* }. (optional)
+--  - requiredKeyItems: Key items required to be able to enter the battlefield - these are removed upon entry (optional)
+--  - title: Title given to players upon victory (optional)
+--  - grantXP: Amount of XP to grant upon victory (optional)
+--  - lossEventParams: Parameters given to the loss event (32002). Defaults to none. (optional)
+function Battlefield:new(data)
```

Comment:
There seems to be a lot of redundant info between the block above and below.  Could this be consolidated and formatted to make it easy to read?